src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+    let initialIndex = 0;
+
+    if (arguments.length < 2 && initialValue !== 0) {
```

Comment:
if there is only 1 argument then initialValue will be undefined anyway, so I would remove the `&& initialValue !== 0`